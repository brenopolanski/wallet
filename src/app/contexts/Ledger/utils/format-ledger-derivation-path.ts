import { LedgerDerivationScheme } from "app/contexts/Ledger/contracts";

export const formatLedgerDerivationPath = (scheme: LedgerDerivationScheme) =>
	`m/${scheme.purpose || 44}'/${scheme.coinType}'/${scheme.account || 0}'/${scheme.change || 0}/${
		scheme.address || 0
	}`;
