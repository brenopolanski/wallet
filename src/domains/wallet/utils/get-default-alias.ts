import { Networks } from "@payvo/sdk";
import { Contracts } from "@payvo/sdk-profiles";

interface GetDefaultAliasInput {
	profile: Contracts.IProfile;
	network: Networks.Network;
}

export const getDefaultAlias = ({ profile, network }: GetDefaultAliasInput): string => {
	const makeAlias = (count: number) => `${network.displayName()} #${count}`;

	const sameCoinWallets = profile.wallets().findByCoinWithNetwork(network.coin(), network.id());

	let sameCoinWalletsCount = sameCoinWallets.length || 1;

	while (profile.wallets().findByAlias(makeAlias(sameCoinWalletsCount))) {
		sameCoinWalletsCount++;
	}

	return makeAlias(sameCoinWalletsCount);
};
