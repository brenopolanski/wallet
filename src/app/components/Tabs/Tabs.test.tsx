import userEvent from "@testing-library/user-event";
import React from "react";
import { render } from "utils/testing-library";

import { Tab, TabList, TabPanel, Tabs } from "./Tabs";

describe("Tabs", () => {
	it("should render", () => {
		const { container, asFragment } = render(
			<Tabs>
				<TabList>
					<Tab tabId={1}>First</Tab>
					<Tab tabId={2}>Second</Tab>
					<Tab tabId={3}>Third</Tab>
				</TabList>
				<div className="mt-5">
					<TabPanel tabId={1}>1</TabPanel>
					<TabPanel tabId={2}>2</TabPanel>
					<TabPanel tabId={3}>3</TabPanel>
				</div>
			</Tabs>,
		);

		expect(container).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render wit counts", () => {
		const { container, asFragment } = render(
			<Tabs>
				<TabList noBackground>
					<Tab tabId={1} count={1}>
						First
					</Tab>
					<Tab tabId={2} count={2}>
						Second
					</Tab>
					<Tab tabId={3} count={3}>
						Third
					</Tab>
				</TabList>
				<div className="mt-5">
					<TabPanel tabId={1}>1</TabPanel>
					<TabPanel tabId={2}>2</TabPanel>
					<TabPanel tabId={3}>3</TabPanel>
				</div>
			</Tabs>,
		);

		expect(container).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render without background", () => {
		const { container, asFragment } = render(
			<Tabs>
				<TabList noBackground>
					<Tab tabId={1}>First</Tab>
					<Tab tabId={2}>Second</Tab>
					<Tab tabId={3}>Third</Tab>
				</TabList>
				<div className="mt-5">
					<TabPanel tabId={1}>1</TabPanel>
					<TabPanel tabId={2}>2</TabPanel>
					<TabPanel tabId={3}>3</TabPanel>
				</div>
			</Tabs>,
		);

		expect(container).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});

	it("should react to use effect call", () => {
		const { container, asFragment, getByTestId } = render(
			<Tabs activeId={2}>
				<TabList>
					<Tab tabId={1}>First</Tab>
					<Tab tabId={2}>Second</Tab>
				</TabList>
				<TabPanel tabId={1}>1</TabPanel>
				<TabPanel tabId={2}>2</TabPanel>
			</Tabs>,
		);

		userEvent.click(getByTestId("tabs__tab-button-1"));

		expect(container).toBeInTheDocument();
		expect(getByTestId("tab-pabel__active-panel")).toHaveTextContent("1");
		expect(asFragment()).toMatchSnapshot();
	});

	it("should handle switching tabs", () => {
		const { getByTestId } = render(
			<Tabs activeId={1}>
				<TabList>
					<Tab tabId={1}>First</Tab>
					<div>separator</div>
					<Tab tabId={2}>Second</Tab>
				</TabList>
				<TabPanel tabId={1}>1</TabPanel>
				<TabPanel tabId={2}>2</TabPanel>
			</Tabs>,
		);

		const firstTab = getByTestId("tabs__tab-button-1");
		const secondTab = getByTestId("tabs__tab-button-2");

		expect(firstTab).toHaveAttribute("aria-selected", "true");
		expect(secondTab).toHaveAttribute("aria-selected", "false");

		firstTab.focus();

		expect(firstTab).toHaveFocus();

		// got right to second tab
		userEvent.keyboard("{arrowright}");

		expect(firstTab).not.toHaveFocus();
		expect(secondTab).toHaveFocus();

		userEvent.keyboard("{enter}");

		expect(getByTestId("tab-pabel__active-panel")).toHaveTextContent("2");

		expect(firstTab).toHaveAttribute("aria-selected", "false");
		expect(secondTab).toHaveAttribute("aria-selected", "true");

		// go right to first tab
		userEvent.keyboard("{arrowright}");

		expect(firstTab).toHaveFocus();
		expect(secondTab).not.toHaveFocus();

		userEvent.keyboard("{space}");

		expect(getByTestId("tab-pabel__active-panel")).toHaveTextContent("1");

		// go left to second tab
		userEvent.keyboard("{arrowleft}");

		expect(firstTab).not.toHaveFocus();
		expect(secondTab).toHaveFocus();

		userEvent.keyboard("{enter}");

		expect(getByTestId("tab-pabel__active-panel")).toHaveTextContent("2");

		// go left to first tab
		userEvent.keyboard("{arrowleft}");

		expect(firstTab).toHaveFocus();
		expect(secondTab).not.toHaveFocus();

		userEvent.keyboard("{space}");

		expect(getByTestId("tab-pabel__active-panel")).toHaveTextContent("1");

		// tab away
		userEvent.tab();
	});
});
