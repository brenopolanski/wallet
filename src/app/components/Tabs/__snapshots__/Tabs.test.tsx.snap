// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should react to use effect call 1`] = `
<DocumentFragment>
  <div>
    <div
      class="Tabs__TabList-sc-1t8sbwb-1 hpBTif"
    >
      <button
        aria-selected="true"
        class="Tabs__TabButton-sc-1t8sbwb-0 ring-focus"
        data-ring-focus-margin="-mx-3"
        data-testid="tabs__tab-button-1"
        role="tab"
        tabindex="0"
        type="button"
      >
        <span>
          First
        </span>
      </button>
      <button
        aria-selected="false"
        class="Tabs__TabButton-sc-1t8sbwb-0 ring-focus"
        data-ring-focus-margin="-mx-3"
        data-testid="tabs__tab-button-2"
        role="tab"
        tabindex="-1"
        type="button"
      >
        <span>
          Second
        </span>
      </button>
    </div>
    <div
      data-testid="tab-pabel__active-panel"
    >
      1
    </div>
  </div>
</DocumentFragment>
`;

exports[`Tabs should render 1`] = `
<DocumentFragment>
  <div>
    <div
      class="Tabs__TabList-sc-1t8sbwb-1 hpBTif"
    >
      <button
        aria-selected="false"
        class="Tabs__TabButton-sc-1t8sbwb-0 ring-focus"
        data-ring-focus-margin="-mx-3"
        data-testid="tabs__tab-button-1"
        role="tab"
        tabindex="-1"
        type="button"
      >
        <span>
          First
        </span>
      </button>
      <button
        aria-selected="false"
        class="Tabs__TabButton-sc-1t8sbwb-0 ring-focus"
        data-ring-focus-margin="-mx-3"
        data-testid="tabs__tab-button-2"
        role="tab"
        tabindex="-1"
        type="button"
      >
        <span>
          Second
        </span>
      </button>
      <button
        aria-selected="false"
        class="Tabs__TabButton-sc-1t8sbwb-0 ring-focus"
        data-ring-focus-margin="-mx-3"
        data-testid="tabs__tab-button-3"
        role="tab"
        tabindex="-1"
        type="button"
      >
        <span>
          Third
        </span>
      </button>
    </div>
    <div
      class="mt-5"
    />
  </div>
</DocumentFragment>
`;

exports[`Tabs should render wit counts 1`] = `
<DocumentFragment>
  <div>
    <div
      class="Tabs__TabList-sc-1t8sbwb-1 kukgZl"
    >
      <button
        aria-selected="false"
        class="Tabs__TabButton-sc-1t8sbwb-0 ring-focus"
        data-ring-focus-margin="-mx-3"
        data-testid="tabs__tab-button-1"
        role="tab"
        tabindex="-1"
        type="button"
      >
        <span>
          First
        </span>
        <span
          class="py-0.5 px-1.5 ml-2 text-sm font-semibold rounded bg-theme-primary-100 dark:bg-theme-secondary-900"
          data-testid="tabs__tab-button-1-count"
        >
          1
        </span>
      </button>
      <button
        aria-selected="false"
        class="Tabs__TabButton-sc-1t8sbwb-0 ring-focus"
        data-ring-focus-margin="-mx-3"
        data-testid="tabs__tab-button-2"
        role="tab"
        tabindex="-1"
        type="button"
      >
        <span>
          Second
        </span>
        <span
          class="py-0.5 px-1.5 ml-2 text-sm font-semibold rounded bg-theme-primary-100 dark:bg-theme-secondary-900"
          data-testid="tabs__tab-button-2-count"
        >
          2
        </span>
      </button>
      <button
        aria-selected="false"
        class="Tabs__TabButton-sc-1t8sbwb-0 ring-focus"
        data-ring-focus-margin="-mx-3"
        data-testid="tabs__tab-button-3"
        role="tab"
        tabindex="-1"
        type="button"
      >
        <span>
          Third
        </span>
        <span
          class="py-0.5 px-1.5 ml-2 text-sm font-semibold rounded bg-theme-primary-100 dark:bg-theme-secondary-900"
          data-testid="tabs__tab-button-3-count"
        >
          3
        </span>
      </button>
    </div>
    <div
      class="mt-5"
    />
  </div>
</DocumentFragment>
`;

exports[`Tabs should render without background 1`] = `
<DocumentFragment>
  <div>
    <div
      class="Tabs__TabList-sc-1t8sbwb-1 kukgZl"
    >
      <button
        aria-selected="false"
        class="Tabs__TabButton-sc-1t8sbwb-0 ring-focus"
        data-ring-focus-margin="-mx-3"
        data-testid="tabs__tab-button-1"
        role="tab"
        tabindex="-1"
        type="button"
      >
        <span>
          First
        </span>
      </button>
      <button
        aria-selected="false"
        class="Tabs__TabButton-sc-1t8sbwb-0 ring-focus"
        data-ring-focus-margin="-mx-3"
        data-testid="tabs__tab-button-2"
        role="tab"
        tabindex="-1"
        type="button"
      >
        <span>
          Second
        </span>
      </button>
      <button
        aria-selected="false"
        class="Tabs__TabButton-sc-1t8sbwb-0 ring-focus"
        data-ring-focus-margin="-mx-3"
        data-testid="tabs__tab-button-3"
        role="tab"
        tabindex="-1"
        type="button"
      >
        <span>
          Third
        </span>
      </button>
    </div>
    <div
      class="mt-5"
    />
  </div>
</DocumentFragment>
`;
